purchasePhone.js

```diff
@@ -2,31 +2,38 @@
         return total*taxRate ;
     }
 
-    function totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney){  
-        var moneyOfAcc = phoneAmount*accesorieyPrice;
-        var moneyOfphone = phoneAmount*phonePrice;
-        var total = moneyOfphone+moneyOfAcc;
-        totalSpendingMoney = total+caculateTax(taxRate,total);   
-        return totalSpendingMoney;
-    }
+    function spendingMoney(phonePrice,accesorieyPrice,phoneAmount,taxRate){     
+        var totalSpendingMoney= phoneAmount*(phonePrice+accesorieyPrice);
+        return totalSpendingMoney + caculateTax(taxRate,totalSpendingMoney); 
+    } 
 
-    function isBankrupt(phonePrice,accesorieyPrice,taxRate,phoneAmount,thresholdSpending, bankMoney,totalSpendingMoney ){  
-    return (((thresholdSpending-totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney))>0)&&((bankMoney-totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney))>0));
+    function isBankrupt(thresholdSpending, bankMoney,allSpendingMoney){  
+    return (((thresholdSpending-allSpendingMoney)>0)&&((bankMoney-allSpendingMoney)>0));
     }
 
-    function main(){
-        var thresholdSpending =4000;
-        var bankMoney = 5000;
+    function buyPhones(thresholdSpending,bankMoney){
+        var phoneAmount = 0;  
+        var allSpendingMoney = 0; 
         var accesorieyPrice = 10;
         var phonePrice = 100;
-        var taxRate = 0.1;
-        var phoneAmount = 0;   
-        var totalSpendingMoney = 0; 
-        while (isBankrupt(phonePrice,accesorieyPrice,taxRate,phoneAmount,thresholdSpending,bankMoney,totalSpendingMoney )){
-            phoneAmount = phoneAmount+ 1; 
+        var taxRate = 0.1; 
+        while (isBankrupt(thresholdSpending,bankMoney,allSpendingMoney)){
+            phoneAmount = phoneAmount+1; 
+            allSpendingMoney = spendingMoney(phonePrice,accesorieyPrice,phoneAmount,taxRate);
+            console.log("the phone number is:" +phoneAmount + "   total spending is :" +allSpendingMoney);
         }
-            totalSpendingMoney = totalSpending(phonePrice,accesorieyPrice,taxRate,phoneAmount,totalSpendingMoney)
-            console.log("the number of phone you have bought is:"+phoneAmount+","+ "the money you have spent(with tax) is:$"+totalSpendingMoney);
+        return {
+        phoneAmount:phoneAmount,
+        allSpendingMoney: spendingMoney(phonePrice,accesorieyPrice,phoneAmount-1,taxRate),
+    };
+    }
+    function main(){
+       
+        var thresholdSpending =4000;
+        var bankMoney = 5000; 
+        var phoneAmount = buyPhones(thresholdSpending,bankMoney).phoneAmount-1;
+        var allSpendingMoney = buyPhones(thresholdSpending,bankMoney).allSpendingMoney;
+        console.log("the number of phone you have bought is:"+(phoneAmount)+","+ "the money you have spent(with tax) is:$"+allSpendingMoney);
```

Comment:
print those information in another function instead of directly call `console.log`